### PR TITLE
Spiderbot misc. fixes

### LINKED ISSUE #15778: Spiderbots can pick up anchored desk bells
#### Difference between expected and actual behavior

Expected: Sanity
Got: Anchored bell can keeps anchored after you drop it

#### Steps to reproduce 

Be spiderbot, go to any desk with a bell, use your pick up verb

#### Issue bingo

Please check whatever applies. More checkboxes checked increase your chances of not being yelled at by every contributor.

- [x] Issue could be reproduced at least once
- [x] Issue could be reproduced by different players
- [x] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/d3athrow/vgstation13/issues)

### PR DESCRIPTION
**Pull request by Pathid**
Closes #15778
~~By the power of copypaste I'm fulfilling anon's request from the thread. This enables spiderbots to use the radio they had inside of them all along.~~ (_Changed, see below._) ~~It only gets the common channel. Still trying to figure out how to make their speech appear in the silicon font.~~

~~I have concerns about letting ventcrawling tattletales run around all the time so I might make the borg radio have to be installed in them instead of spawning on creation, I don't know yet.~~

Edit: See https://github.com/vgstation-coders/vgstation13/pull/17446#issuecomment-365713469, I'm calling off giving them the ability to transmit since I think being able to radio from inside vents would be too strong.

:cl:
* rscadd: Spider-bot speech is in the "siliconsay" style, has the robotic speech bubble and tcomms shows them as "machine" instead of "unknown."
* rscadd: Spider-bots have a verb to turn their radio listening on and off.
* tweak: Spider-bots' ability to launch certain "fun" items works and launches properly.
* bugfix: Fixed spider-bots being able to pick up anchored objects.
* tweak: Spider-bots take damage from EMPs and have a small chance to explode.

### CODE REVIEW

**Comment by Kurfursten** [comment]
IIRC you want to override get_message_mode to `return MODE_ROBOT`, then to change the speech bubble, override /say to `return ..(message, "R")`

You may need to do some testing on this

**Comment by Pathid** [comment]
Didn't I already do that? I've got the bubble working but the font is still normal.

Also after rethinking this, the radio did serve a purpose before. It allowed them to listen. And with the fact that they were capable of speaking on intercoms with :i might have given them enough comms capability already.

**Comment by Pathid** [comment]
Alright, figured it out.

**Comment by Pathid** [comment]
Wait I can make this better.

**Comment by jknpj** [comment]
You might want to take a look at #15778

File: code/modules/mob/living/simple_animal/friendly/spiderbot.dm [hunk 1]
```diff
@@ -235,8 +239,19 @@
...
 		var/obj/item/weapon/grenade/G = held_item
-		G.forceMove(src.loc)
-		G.prime()
+
+		//Make a dumbfire throw
+		var/turf/lob_target
+		var/lob_dir = dir
+		var/turf/start_turf = get_turf(src)
+		var/current_turf = start_turf
+		for(var/i = 1; i <= lob_range; i++)
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/friendly/spiderbot.dm]: use `for(var/i in 1 to lob_range)`

File: code/modules/mob/living/simple_animal/friendly/spiderbot.dm [hunk 2]
```diff
@@ -287,3 +305,21 @@
...
+/mob/living/simple_animal/spiderbot/verb/Toggle_Listening()
+	set name = "Toggle Listening"
+	set desc = "Toggle listening channel on or off."
+	set category = "Spiderbot"
+
+	if(stat)
+		to_chat(src, "Can't do that while incapacitated or dead.")
+
+	radio.listening = !radio.listening
+	to_chat(src, "<span class='notice'>Radio is [radio.listening ? "" : "no longer"] receiving broadcasts.</span>")
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/friendly/spiderbot.dm]: this will result in an extra space character that I cannot unsee

**Review by DamianX — CHANGES_REQUESTED** [review]

**Comment by DamianX** [comment]
Do you think players reading the changelog will have any clue what the MOB_ROBOTIC flag does?

**Comment by Probe1** [comment]
Yeah keep in mind what I said last time.  You don't need to include backend and minor bug fixes for the changelog, and it's best to condense stuff.  Additions with additions instead of a new line for every **related** thing.  

If Joe Cantcode sees a big list of shit that he doesn't understand or doesn't matter to his experience then his eyes slide right on by it all.

File: code/modules/mob/living/simple_animal/friendly/spiderbot.dm [hunk 3]
```diff
@@ -287,3 +322,21 @@
...
+/mob/living/simple_animal/spiderbot/treat_speech(var/datum/speech/speech, genesay = 0)
+	..(speech)
+	speech.message_classes.Add("siliconsay")
+
+/mob/living/simple_animal/spiderbot/verb/Toggle_Listening()
+	set name = "Toggle Listening"
+	set desc = "Toggle listening channel on or off."
+	set category = "Spiderbot"
+
+	if(stat)
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/friendly/spiderbot.dm]: use `incapacitated()`

File: code/modules/mob/living/simple_animal/friendly/spiderbot.dm [hunk 4]
```diff
@@ -287,3 +322,21 @@
...
+	..(speech)
+	speech.message_classes.Add("siliconsay")
+
+/mob/living/simple_animal/spiderbot/verb/Toggle_Listening()
+	set name = "Toggle Listening"
+	set desc = "Toggle listening channel on or off."
+	set category = "Spiderbot"
+
+	if(incapacitated())
+		to_chat(src, "Can't do that while incapacitated or dead.")
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/friendly/spiderbot.dm:339]: Needs a `return` after this

Comment (Pathid) [reply code/modules/mob/living/simple_animal/friendly/spiderbot.dm:339]: Good spot.

**Review by DamianX — APPROVED** [review]

**Pull request merged**